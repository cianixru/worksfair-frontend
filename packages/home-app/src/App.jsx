import React, { Component } from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { PersistGate } from 'redux-persist/integration/react';
import { ToastContainer } from 'react-toastify';

import 'font-awesome/css/font-awesome.min.css';
import { store, persistor } from './store';
import 'bulma/css/bulma.min.css';
import 'react-toastify/dist/ReactToastify.min.css';
import 'react-responsive-carousel/lib/styles/carousel.min.css';
import './styles/Index.scss';

import Routes from './routes/Index';

class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <PersistGate loading={null} persistor={persistor}>
          <div className="App">
            <Router>
              <Routes />
            </Router>
            <ToastContainer />
          </div>
        </PersistGate>
      </Provider>
    );
  }
}

export default App;
