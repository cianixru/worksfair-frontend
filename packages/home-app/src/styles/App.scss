@import './Dashboard.scss';

* {
    -webkit-box-sizing: border-box;
    -moz-box-sizing: border-box;
    box-sizing: border-box;
}

html, body {
    height: 100%;
    margin: 0;
}

body, h1, h2, h3, h4, h5, h6,
input[type="text"],
input[type="password"],
input[type="email"],
textarea, select,
input[type="button"],
input[type="submit"],
button {
    font-family: raleway, helveticaneue, helvetica neue, Helvetica, Arial, sans-serif;
    text-transform: none;
}

body, .dashboard-container, .main-container {
    font-size: 16px;
    min-height: 80vh;
    width: 100%;
}

.footer {
    position: relative;
    width: 100%;
    bottom: 0px;
    left: 0px;
    padding-bottom: 3em;
}

.field {
    padding-bottom: 15px;
}

.auth-layout {
    padding: 20px;
    margin: 5px 0px 0px 0px;
    background-image: url('../assets/main-search-video-poster.jpg');
    min-height: 93vh;
}

.auth-content {
    padding: 30px;
    border-radius: 5px;
    background-color: rgba(250, 250, 250, 0.9);
}

.auth-overlay {
    position: fixed;
    top: 20px;
    left: 5px;
    align-items: center;
    justify-content: center;
    box-sizing: border-box;
    background-color: rgba(0, 0, 0, 0.5);
    width: 102vw;
    height: 100vh;
}

@media(max-width: 767px) {
    .signup {
        margin-top: 30%;
        position: relative;
        top: 120px;
    }
}

@media(max-width: 991px) {
    .auth-overlay {
        overflow: scroll;
        padding: 5%;
    }
}

.input-error {
    position: relative;
    display: block;
    top: 2px;
    left: 0.5em;
    color: red;
    font-size: 14px;
}

.spin-loader {
    position: fixed;
    background-color: rgba(0, 0, 0, 0.3);
    height: 100%;
    width: 100%;
    z-index: 100000;
}

.spinner {
    position: absolute;
    top: 40%;
    left: 50%;
    margin: 0 auto;
    z-index: 100000;
}

.is-180x180 {
    height: 180px;
    width: 180px;
}

.is-180x180 img {
    max-height: 100%;
    max-width: 100%;
}

.is-150x150 {
    height: 150px;
    width: 150px;
}

.is-150x150 img {
    max-height: 100%;
    max-width: 100%;
}

img {
    object-fit: cover;
}

audio, canvas, progress, video {
    display: inline-block;
    vertical-align: baseline;
}

.margin-right-7 {
  margin-right: 7px;
}

.margin-top-25 {
  margin-top: 25px;
}

.margin-top-15-vh {
  margin-top: 15vh;
}

.has-text-extra-large {
    margin-top: -100px;
    font-size: 125px;
}

.has-text-large {
  font-size: 50px;
}

.auth {
  border-left: 0px;
  border-top: 0px;
  border-right: 0px;
  box-shadow: none;
  margin-bottom: 20px;
}

.padding-top-25 {
    padding-top: 25px;
}
