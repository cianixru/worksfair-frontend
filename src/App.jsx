import React, { Component } from 'react';

import 'bulma/css/bulma.min.css';
import { library } from '@fortawesome/fontawesome-svg-core';
import { faPlusCircle } from '@fortawesome/free-solid-svg-icons';

import Footer from './components/Footer/Footer';
import Navbar from './components/Navbar/Navbar';

library.add(faPlusCircle);

class App extends Component {
  render() {
    return (
      <div className="App">
        <header>
          <Navbar />
          
        </header>

        <Footer />
      </div>
    );
  }
}

export default App;
