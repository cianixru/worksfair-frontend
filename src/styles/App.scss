@import './Dashboard.scss';

* {
    -webkit-box-sizing: border-box;
    -moz-box-sizing: border-box;
    box-sizing: border-box;
}

html, body {
    height: 100%;
    margin: 0;
}

body, h1, h2, h3, h4, h5, h6,
input[type="text"],
input[type="password"],
input[type="email"],
textarea, select,
input[type="button"],
input[type="submit"],
button {
    font-family: raleway, helveticaneue, helvetica neue, Helvetica, Arial, sans-serif;
    text-transform: none;
}

body {
    font-size: 16px;
}

.footer {
    position: relative;
    width: 100%;
    bottom: 0em;
    left: 0em;  
    padding-bottom: 3em;
}

.field {
    padding-bottom: 15px;
}

.layout-auth {
    padding: 20px;
    margin: 30px 0px;
}

.input-error {
    position: relative;
    display: block;
    top: 2px;
    left: 0.5em;
    color: red;
    font-size: 14px;
}

.spin-loader {
    position: fixed;
    background-color: rgba(0, 0, 0, 0.3);
    height: 100%;
    width: 100%;
    z-index: 100000;
}

.spinner {
    position: absolute;
    top: 40%;
    left: 50%;
    margin: 0 auto;
    z-index: 100000;
}
