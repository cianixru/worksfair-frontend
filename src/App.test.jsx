import React from 'react';
import { render } from 'react-testing-library';

import App from './App';

test('Test that the Footer displays', () => {
  const component = render(
    <App />
  );
  expect(component).toMatchSnapshot();
});
