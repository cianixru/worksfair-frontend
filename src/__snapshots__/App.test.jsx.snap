// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test that the Footer displays 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div>
          <div
            class=""
          />
          <header>
            <nav
              aria-label="main navigation"
              class="navbar"
              data-testid=""
              role="navigation"
            >
              <div
                class="navbar-brand"
              >
                <a
                  class="navbar-item"
                  href="/home"
                >
                  <img
                    alt="logo"
                    height="28"
                    src="logo.png"
                    width="112"
                  />
                </a>
                <a
                  aria-expanded="false"
                  aria-label="menu"
                  class="navbar-burger burger"
                  data-target="navbarBasicExample"
                  href="/"
                  role="button"
                >
                  <span
                    aria-hidden="true"
                  />
                  <span
                    aria-hidden="true"
                  />
                  <span
                    aria-hidden="true"
                  />
                </a>
              </div>
              <div
                class="navbar-menu"
              >
                <div
                  class="navbar-start"
                >
                  <a
                    class="navbar-item"
                    href="/"
                  >
                    Home
                  </a>
                  <a
                    class="navbar-item"
                    href="/"
                  >
                    About
                  </a>
                </div>
                <div
                  class="navbar-end"
                >
                  <div
                    class="navbar-item"
                  >
                    <div
                      class="buttons"
                    >
                      <a
                        class="button is-white"
                        href="/login"
                      >
                        Log in
                      </a>
                      <a
                        class="button is-white"
                        href="/signup"
                      >
                        Sign up
                      </a>
                      <a
                        class="button is-danger is-outlined is-rounded"
                        href="/"
                      >
                        Create Free Webpage
                      </a>
                    </div>
                  </div>
                </div>
              </div>
            </nav>
          </header>
          <div
            class="main-search-container dark-overlay"
          >
            <div
              class="main-search-inner"
            >
              <div
                class="container"
              >
                <div
                  class="columns"
                >
                  <div
                    class="is-12"
                  >
                    <h2>
                      Find Businesses and Services Near You.
                    </h2>
                    <h4>
                      Discover top-ranked businesses and services that are ready to serve your needs.
                    </h4>
                    <form>
                      <div
                        class="main-search-input"
                      >
                        <div
                          class="main-search-input-item"
                        >
                          <input
                            class="input"
                            name="keywords"
                            placeholder="What do you need?"
                            type="text"
                            value=""
                          />
                        </div>
                        <div
                          class="main-search-input-item location"
                        >
                          <input
                            class="input"
                            name="location"
                            placeholder="Which city or town?"
                            type="text"
                            value=""
                          />
                        </div>
                        <button
                          class="button is-info is-rounded"
                          disabled=""
                        >
                          Search
                        </button>
                      </div>
                    </form>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="video-container"
            >
              <video
                autoplay=""
                loop=""
                poster="main-search-video-poster.jpg"
              >
                <source
                  src="main-search-video.mp4"
                  type="video/mp4"
                />
              </video>
            </div>
          </div>
        </div>
        <div
          class="Toastify"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div>
        <div
          class=""
        />
        <header>
          <nav
            aria-label="main navigation"
            class="navbar"
            data-testid=""
            role="navigation"
          >
            <div
              class="navbar-brand"
            >
              <a
                class="navbar-item"
                href="/home"
              >
                <img
                  alt="logo"
                  height="28"
                  src="logo.png"
                  width="112"
                />
              </a>
              <a
                aria-expanded="false"
                aria-label="menu"
                class="navbar-burger burger"
                data-target="navbarBasicExample"
                href="/"
                role="button"
              >
                <span
                  aria-hidden="true"
                />
                <span
                  aria-hidden="true"
                />
                <span
                  aria-hidden="true"
                />
              </a>
            </div>
            <div
              class="navbar-menu"
            >
              <div
                class="navbar-start"
              >
                <a
                  class="navbar-item"
                  href="/"
                >
                  Home
                </a>
                <a
                  class="navbar-item"
                  href="/"
                >
                  About
                </a>
              </div>
              <div
                class="navbar-end"
              >
                <div
                  class="navbar-item"
                >
                  <div
                    class="buttons"
                  >
                    <a
                      class="button is-white"
                      href="/login"
                    >
                      Log in
                    </a>
                    <a
                      class="button is-white"
                      href="/signup"
                    >
                      Sign up
                    </a>
                    <a
                      class="button is-danger is-outlined is-rounded"
                      href="/"
                    >
                      Create Free Webpage
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </nav>
        </header>
        <div
          class="main-search-container dark-overlay"
        >
          <div
            class="main-search-inner"
          >
            <div
              class="container"
            >
              <div
                class="columns"
              >
                <div
                  class="is-12"
                >
                  <h2>
                    Find Businesses and Services Near You.
                  </h2>
                  <h4>
                    Discover top-ranked businesses and services that are ready to serve your needs.
                  </h4>
                  <form>
                    <div
                      class="main-search-input"
                    >
                      <div
                        class="main-search-input-item"
                      >
                        <input
                          class="input"
                          name="keywords"
                          placeholder="What do you need?"
                          type="text"
                          value=""
                        />
                      </div>
                      <div
                        class="main-search-input-item location"
                      >
                        <input
                          class="input"
                          name="location"
                          placeholder="Which city or town?"
                          type="text"
                          value=""
                        />
                      </div>
                      <button
                        class="button is-info is-rounded"
                        disabled=""
                      >
                        Search
                      </button>
                    </div>
                  </form>
                </div>
              </div>
            </div>
          </div>
          <div
            class="video-container"
          >
            <video
              autoplay=""
              loop=""
              poster="main-search-video-poster.jpg"
            >
              <source
                src="main-search-video.mp4"
                type="video/mp4"
              />
            </video>
          </div>
        </div>
      </div>
      <div
        class="Toastify"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
